spec/support/idv_examples/cancel_at_idv_step.rb

```diff
@@ -19,20 +19,54 @@
     expect(current_path).to eq(original_path)
   end
 
-  it 'shows the user a cancellation message with the option to cancel and reset idv' do
-    click_link t('links.cancel')
+  context 'with an sp' do
+    it 'shows the user a cancellation message with the option to cancel and reset idv' do
+      failure_to_proof_url = 'https://www.example.com/failure'
+      sp_name = 'Test SP'
+      allow_any_instance_of(SessionDecorator).to receive(:failure_to_proof_url).
+        and_return(failure_to_proof_url)
+      allow_any_instance_of(ServiceProviderSessionDecorator).to receive(:failure_to_proof_url).
+        and_return(failure_to_proof_url)
+      allow_any_instance_of(SessionDecorator).to receive(:sp_name).and_return(sp_name)
+      allow_any_instance_of(ServiceProviderSessionDecorator).to receive(:sp_name).
+        and_return(sp_name)
 
-    expect(page).to have_content(t('idv.cancel.modal_header'))
-    expect(current_path).to eq(idv_cancel_path)
+      click_link t('links.cancel')
 
-    click_on t('forms.buttons.cancel')
+      expect(page).to have_content(t('idv.cancel.modal_header'))
+      expect(current_path).to eq(idv_cancel_path)
 
-    expect(page).to have_content(t('headings.cancellations.confirmation'))
-    expect(current_path).to eq(idv_cancel_path)
+      click_on t('forms.buttons.cancel')
+
+      expect(page).to have_content(t('headings.cancellations.confirmation'))
+      expect(current_path).to eq(idv_cancel_path)
+
+      expect(page).to have_link("‹ #{t('links.back_to_sp', sp: sp_name)}",
+                                href: failure_to_proof_url)
+
+      # After visiting /verify, expect to redirect to the jurisdiction step,
+      # the first step in the IdV flow
+      visit idv_path
+      expect(current_path).to eq(idv_jurisdiction_path)
+    end
+  end
+
+  context 'without an sp' do
+    it 'shows the user a cancellation message with the option to cancel and reset idv' do
+      click_link t('links.cancel')
+
+      expect(page).to have_content(t('idv.cancel.modal_header'))
+      expect(current_path).to eq(idv_cancel_path)
+
+      click_on t('forms.buttons.cancel')
+
+      expect(page).to have_content(t('headings.cancellations.confirmation'))
```

Comment:
I thought we were going to provide a link to go back to the account page in this scenario.

Reply:
Also, we want to make sure that in this scenario, there aren't any links to go back to the SP, but because this is a shared example, this test will fail in some cases. The solution is to skip this test if there is an SP defined, by using [RSpec's conditional filters](https://relishapp.com/rspec/rspec-core/docs/filtering/conditional-filters):
```ruby
context 'without an sp' do
    it 'shows the user a cancellation message', if: sp.nil? do
      ...
    end
end
```

Reply:
Done done.